featurebyte/api/use_case_or_context_mixin.py

```diff
@@ -0,0 +1,153 @@
+"""
+Base UseCase and Context mixin
+"""
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+from typeguard import typechecked
+
+from featurebyte.api.api_object import ApiObject
+from featurebyte.api.api_object_util import iterate_api_object_using_paginated_routes
+from featurebyte.api.observation_table import ObservationTable
+from featurebyte.models.base import PydanticObjectId
+
+
+class UseCaseOrContextMixin(ApiObject):
+    """
+    Base UseCase and Context mixin
+    """
+
+    default_preview_table_id: Optional[PydanticObjectId]
+    default_eda_table_id: Optional[PydanticObjectId]
+
+    @property
+    def default_eda_table(self) -> Optional[ObservationTable]:
+        """
+        Returns the target object of the UseCase or Context.
+
+        Returns
+        -------
+        ObservationTable
+            The ObservationTable object of the UseCase or Context.
+        """
+        if self.default_eda_table_id:
+            return ObservationTable.get_by_id(self.default_eda_table_id)
+
+        return None
+
+    @property
+    def default_preview_table(self) -> Optional[ObservationTable]:
+        """
+        Returns the context object of the UseCase or Context.
+
+        Returns
+        -------
+        ObservationTable
+            The ObservationTable object of the UseCase or Context.
+        """
+        if self.default_preview_table_id:
+            return ObservationTable.get_by_id(self.default_preview_table_id)
+
+        return None
+
+    @typechecked
+    def add_observation_table(self, observation_table_name: str) -> None:
+        """
+        Add observation table for the UseCase or Context.
+
+        Parameters
+        ----------
+        observation_table_name: str
+            New observation table to be added.
+
+        Examples
+        --------
+        >>> use_case = catalog.get_use_case("use_case")
+        >>> use_case.add_observation_table(observation_table_name)  # doctest: +SKIP
+        """
+        observation_table = ObservationTable.get(observation_table_name)
+        observation_table.update(update_payload={"context_id": self.id}, allow_update_local=False)
```

Comment:
how do we know that the `self.id` corresponds to a context? this operation feels a little dangerous as it feels like we could set any id into this `context_id` field.

Reply:
I will move it into the Context api object